cac-agent/src/main/java/com/moesol/cac/agent/selector/AbstractSelectorKeyManager.java

```diff
@@ -234,17 +245,54 @@ private String maybeUseDefaultCertificateName(Principal[] issuers) {
 			if (ks == null) {
 				return new String[0];
 			}
+			Set<Principal> issuerSet = null;
+			if (issuers != null && issuers.length > 0) {
```

Comment:
These changes look fine. I'm going to through out some suggestions and you can let me know what you think. They are in the spirit of trying to simplify the code/remove branches.

I'm pretty sure it is safe to create an empty HashSet when issues is `[]`. We can also handle the issuers = null case by creating an empty HashSet. Then later the code gets simpler because we will not have to check of issuerSet is null.

Maybe something like `issuerSet = new HashSet<>(issuers == null ? [] : issuers)`

Reply:
During a TLS handshake, sometimes the server will provide a list of acceptable issuers and sometimes it won't. If it doesn't provide a list, we shouldn't do any issuer filtering.

At the moment, I'm using `issuers == null` as a state invariant to mean "no issuer list was provided by the server." Your suggestion is to use `issuers.isEmpty()` as that state invariant instead. Either will work just fine.

But this change won't make the later code simpler; we always must to check that state invariant (`null` or `isEmpty()`) because it triggers a different behavior (disabling the issue filtering).

I liked the use of null because it was more explicit in this regard, forcing later code to check the state invariant. This reduces the chances of a future developer checking for items in the empty HashSet when they shouldn't be checking at all. But I'll absolutely defer to your preference.

Reply:
Sounds good.